napari/_vispy/volume.py

```diff
@@ -58,6 +58,23 @@
 
 frag_dict['minip'] = MINIP_FRAG_SHADER
 
+AVG_SNIPPETS = dict(
+    before_loop="""
+        float sum = 0.0; // The sum of encountered values
+        """,
+    in_loop="""
+        sum += val;
+        """,
+    after_loop="""
+        // Calculate mean value
+        float meanval = 0.0;  // The average value encountered
+        meanval = sum / float(nsteps);
```

Comment:
```suggestion
        float meanval = sum / float(nsteps);  // The average value encountered
```

I think you can declare it and assign it at the same time?

Reply:
That should make my laptop fans whir ever so slightly less!